istioctl/cmd/sidecar-bootstrap.go

```diff
@@ -0,0 +1,643 @@
+	return fmt.Sprintf("%x", sha256.Sum256([]byte(address)))[0:7]
+}
+
+func deriveDiscoveryAddressPort(service string, namespace string, kubeClient kubernetes.Interface) (string, error) {
+	foundService, err := kubeClient.CoreV1().Services(namespace).Get(context.TODO(), service, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, port := range foundService.Spec.Ports {
+		if strings.HasPrefix(port.Name, "tcp-istiod") {
+			return string(port.NodePort), nil
+		}
+	}
+	return "", fmt.Errorf("failed to find tcp-istiod port")
+}
+
+func copyCertificates(kubeClient kubernetes.Interface,
+	addressCertMapping map[string]workloadEntryAddressKeys) error {
+	var callback ssh.HostKeyCallback
+	if sshIgnoreHostKeys {
+		callback = ssh.InsecureIgnoreHostKey()
+	} else {
+		user, err := user.Current()
+		if err != nil {
+			return err
+		}
+		callback, err = knownhosts.New(path.Join(user.HomeDir, ".ssh", "known_hosts"))
+		if err != nil {
+			return err
+		}
+	}
+
+	for address, certs := range addressCertMapping {
+		sshConfig := &ssh.ClientConfig{
+			User:            sshUser,
+			Auth:            []ssh.AuthMethod{sshAuthMethod},
+			HostKeyCallback: callback,
+		}
+		client, err := ssh.Dial("tcp", address+":22", sshConfig)
+		if err != nil {
+			return err
+		}
+		// Ensure the remote directory exists.
+		mkdirErr := remoteRunCommand("mkdir -p "+remoteDirectory, client)
+		if mkdirErr != nil {
+			client.Close()
+			return err
+		}
+
+		remoteCaPath := path.Join(remoteDirectory, "ca-cert-"+address+".pem")
+		err = remoteCopyFile(certs.CaCert, remoteCaPath, client)
+		if err != nil {
+			client.Close()
+			return err
+		}
+
+		remoteCertPath := path.Join(remoteDirectory, "cert-"+address+".pem")
+		err = remoteCopyFile(certs.Cert, remoteCertPath, client)
+		if err != nil {
+			client.Close()
+			return err
+		}
+		remoteKeyPath := path.Join(remoteDirectory, "key-"+address+".pem")
+		err = remoteCopyFile(certs.Key, remoteKeyPath, client)
+		if err != nil {
+			client.Close()
+			return err
+		}
+
+		if startIstio {
+			addressIdentifier := addressToPodNameAddition(address)
+			istiodPort, err := deriveDiscoveryAddressPort(certs.ServiceName, certs.ServiceNamespace, kubeClient)
+			if err != nil {
+				client.Close()
+				return err
+			}
+			discoveryAddress := "istiod.istio-system.svc:" + istiodPort
+			var authPolicyStr string
+			if mutualTLS {
+				authPolicyStr = "MUTUAL_TLS"
+			} else {
+				authPolicyStr = "NONE"
+			}
+
+			meshPath := path.Join(remoteDirectory, "mesh.yaml")
+			err = remoteCopyFile([]byte(`---
+defaultConfig:
+	serviceCluster:`+certs.ServiceNamespace+`
+	controlPlaneAuthPolicy: `+authPolicyStr+`
+	discoveryAddress: `+discoveryAddress), meshPath, client)
+			if err != nil {
+				client.Close()
+				return err
+			}
+
+			//you need to deal with Sidecar CR if you want it to be "non-captured" mode
+			err = remoteRunCommand(
+				"docker run -d --name istio-proxy --network host "+
```

Comment:
The istio.io documentation at https://istio.io/docs/reference/commands/istioctl/ will be generated from the command.  This command requires Docker on the workload.  Please add a `Long:` description that indicates what the command requires and really does.  Something like "Uses Docker to run the Istio sidecar, allowing the workload to participate in an Istio mesh."